apps/server/src/modules/synchronization/uc/synchronization.uc.ts

```diff
@@ -0,0 +1,108 @@
+import { UserService } from '@modules/user';
+import { Injectable } from '@nestjs/common';
+import { Logger } from '@src/core/logger';
+import { SanisResponse, SchulconnexRestClient } from '@src/infra/schulconnex-client';
+import { ConfigService } from '@nestjs/config';
+import { SynchronizationService } from '../domain/service';
+import { Synchronization } from '../domain';
+import { SynchronizationStatusModel } from '../domain/types';
+import { SynchronizationLoggable } from '../domain/loggable';
+import { SynchronizationErrorLoggableException } from '../domain/loggable-exception';
+import { SynchronizationConfig } from '../config';
+
+@Injectable()
+export class SynchronizationUc {
+	constructor(
+		private readonly configService: ConfigService<SynchronizationConfig, true>,
+		private readonly schulconnexRestClient: SchulconnexRestClient,
+		private readonly synchronizationService: SynchronizationService,
+		private readonly userService: UserService,
+		private readonly logger: Logger
+	) {
+		this.logger.setContext(SynchronizationUc.name);
+	}
+
+	public async updateSystemUsersLastSyncedAt(systemId: string): Promise<void> {
+		this.logger.info(new SynchronizationLoggable('Start synchronization users from systemId', systemId));
+
+		const synchronizationId = await this.synchronizationService.createSynchronization();
+
+		try {
+			const usersToCheck = await this.findUsersToSynchronize(systemId);
+			const chunkSize = this.configService.get<number>('SYNCHRONIZATION_CHUNK');
+			const chunks = this.chunkArray(usersToCheck, chunkSize);
+			const promises = chunks.map((chunk) => this.updateLastSyncedAt(chunk, systemId));
+			const results = await Promise.all(promises);
+			const userSyncCount = results.reduce((acc, curr) => +acc + +curr, 0);
+
+			await this.updateSynchronization(synchronizationId, SynchronizationStatusModel.SUCCESS, userSyncCount);
+			this.logger.info(new SynchronizationLoggable('End synchronization users from systemId', systemId, userSyncCount));
+		} catch (error) {
+			let logMessage = '';
+			if (error instanceof SynchronizationErrorLoggableException) {
+				logMessage = `${error.getLogMessage()?.data?.errorMessage as string}: ${
+					error.getLogMessage()?.data?.systemId as string
+				}`;
+			} else {
+				logMessage = `Synchronisation process failed for users provided by the system ${systemId}`;
```

Comment:
one of the intentions of the loggable is to put the construction of the string inside the loggable. This ensures control over the data that is logged, for data security purposes. Having a generic loggable for your module that you pass any string with injected data into, defeats the point.

please change this logic to have the text specific to your loggable, with one loggable per relevant case (ie a seperate loggable for this message).
When you get a SynchronizationErrorLoggableException here, you should be able to directly pass error.getLogMessage() into updateSynchronization, without further transformation logic. In your else block, you can simply do something like `SynchronisationProcessUnknownError(systemid, error), and then proceed in the same way (call update with loggable.getLogMessage(), and log it)

Reply:
changed